packages/async-flow/src/replay-membrane.js

```diff
@@ -258,11 +311,21 @@ export const makeReplayMembrane = (
   };
   harden(makeGuestForHostRemotable);
 
-  const makeGuestForHostVow = hVow => {
-    // TODO in order to support E *well*,
-    // use HandledPromise to make `promise` a handled promise for hVow
-    const { promise, resolve, reject } = makePromiseKit();
-    guestPromiseMap.set(promise, harden({ resolve, reject }));
+  /**
+   * @param {Vow} hVow
+   * @param {Promise} [promiseKey]
+   *   If provided, use this promise as the key in the guestPromiseMap
+   *   rather than the returned promise. This only happens when the
+   *   promiseKey ends up forwarded to the returned promise anyway, so
+   *   associating it with this resolve/reject pair is not incorrect.
+   *   It is needed when `promiseKey` is also entered into the bijection
+   *   paired with hVow.
```

Comment:
Let's leave the `promiseKey` stuff out of this PR for now.

Reply:
Done